contentful_cli/management.py

```diff
@@ -393,7 +419,12 @@ def construct_base_url(self):
         raise Exception('No suitable contentful-api-gateway endpoint for Contentful subdomain {}'.format(self.subdomain))
 
     def invoke(self, arguments, session, oauth_token, gateway_api_key, run=True):
-        expanded_path = self.template.format(**arguments)
+
+        templating_arguments = dict(arguments)
+        if self.environment_aware and arguments.get('environment_id'):
+            templating_arguments['space_id'] += '/environments/' + arguments['environment_id']
```

Comment:
This feels super hacky. Instead of modifying the endpoint templates (which had to be modified to flag them as environment capable anyway), this hotpatches another unrelated field to add the environment. It makes a possibly dangerous assumption about the ordering of the fields in the endpoint URL and could break with future endpoints.

We could make this clearer while maintaining the ability to not specify an environment in API calls (for accounts that haven't been migrated) by adding a `{environment}` templating argument that is `""` when an environment isn't used/set, but is `'/environments/' + arguments['environment_id']` when it is used.

Reply:
I was uncomfortable with this too. I considered what you suggest here, but backed off because it made the template strings extremely unwieldy without clear benefit.

I disagree that the space id is an "unrelated" field - the space and the environment are inextricably bound together. Also I don't see how the assumption about field order is "dangerous". But I would concede that the way it's used does not make it clear that it is a special case. How would you feel about replacing this with something that makes it clear it is getting special treatment, such as replacing this:

`/spaces/{space_id}/entries/{entry_id}`

With:

`{environment_specifier}/entries/{entry_id}`

Where we document `{environment_specifier}` as being one of:

```
spaces/{space_id}
spaces/{space_id}/environments/{environment_id}
```

...dependent on whether the endpoint is flagged as environment-aware and whether an environment is specified?

(Note also that it is *always* valid not to specify an environment for merely environment-aware commands. This implies use of the master environment. I think you're already clear on that but I want to mention it in case I'm not being clear myself.)